opm/simulators/thresholdPressures.hpp

```diff
@@ -201,21 +201,22 @@ void computeMaxDp(std::map<std::pair<int, int>, double>& maxDp,
             int pvtRegionIdx = pvtRegion[cellIdx];
 
             double T = initialState.temperature()[cellIdx];
-            double p = initialState.pressure()[cellIdx];
+            double p = phasePressure[opos][cellIdx];
             double Rs = initialState.gasoilratio()[cellIdx];
             double RsSat = pvto.saturatedGasDissolutionFactor(pvtRegionIdx, T, p);
 
+            double b;
             if (Rs >= RsSat) {
-                double b = pvto.saturatedInverseFormationVolumeFactor(pvtRegionIdx, T, p);
-                rho[opos][cellIdx] = surfaceDensity[pvtRegionIdx][opos]*b;
+                b = pvto.saturatedInverseFormationVolumeFactor(pvtRegionIdx, T, p);
             }
             else {
-                double b = pvto.inverseFormationVolumeFactor(pvtRegionIdx, T, p, Rs);
-                rho[opos][cellIdx] = surfaceDensity[pvtRegionIdx][opos]*b;
-                if (pu.phase_used[BlackoilPhases::Vapour]) {
-                    int gpos = pu.phase_pos[BlackoilPhases::Vapour];
-                    rho[opos][cellIdx] += surfaceDensity[pvtRegionIdx][gpos]*Rs*b;
-                }
+                b = pvto.inverseFormationVolumeFactor(pvtRegionIdx, T, p, Rs);
+            }
+
+            rho[opos][cellIdx] = surfaceDensity[pvtRegionIdx][opos]*b;
+            if (pu.phase_used[BlackoilPhases::Vapour]) {
+                int gpos = pu.phase_pos[BlackoilPhases::Vapour];
+                rho[opos][cellIdx] += surfaceDensity[pvtRegionIdx][gpos]*Rs*b;
```

Comment:
> Should not Rs be RsSat here if we are in the saturated regime? 

isn't this branch also taken if gas is enabled in principle but just not present for the current cell?

> I assume it will most likely be explicitly set to be equal to RsSat, but still.

the simulator does so if I remember correctly. The EQUIL initializer might not do it. (But I could be wrong.)

>  I believe this was one of the reasons this was not simplified originally?

yeah, I did not have a sufficient overview of how the main part of the simulator does this, so I just took guesses. (I'd argue that both views are reasonable.)